### PR TITLE
Allow camera_params update through `update_items_metadata`

### PR DESCRIPTION
**Pull request by jean-lucas**
User can now directly update the `dataset_item.camera_params` field though the call to `dataset.update_items_metadata(mapping)`

Works alongside the [backend PR](https://github.com/scaleapi/scaleapi/pull/38112) - will be merged before this one.

To test.

```python
params = {
        "position": {"x": 0, "y": 0, "z": 1},
        "heading": {"w": 1,"x": 0.5,"y":0.5,"z": 0.5},
        "fx": 2, "fy": 2,
        "cx": 2, "cy": 2,
        "p1": 1, "k2": 2,

}

dataset.update_items_metadata({'some_ref': {'camera_params': params}})
```

### CODE REVIEW

File: nucleus/camera_params.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+from dataclasses import dataclass
```
Comment (jean-lucas) [inline nucleus/camera_params.py:1]: copied from `nucleus/dataset_item.py`, no logic changed.

File: nucleus/quaternion.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass
```
Comment (jean-lucas) [inline nucleus/quaternion.py:1]: copied from `nucleus/dataset_item.py` no logic changed.

File: nucleus/camera_params.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                raise ValueError(
+                    f'Invalid Camera Model, the supported options are "{CameraModels.BROWN_CONRADY}" and "{CameraModels.FISHEYE}"'
+                )
+
+    @classmethod
+    def from_json(cls, payload: Dict[str, Any]):
+        """Instantiates camera params object from schematized JSON dict payload."""
+        keys = set(payload.keys())
+        if not keys.issuperset(REQUIRED_CAMERA_PARAMS_KEYS):
+            raise ValueError(f'The following fields must be present in the camera_params dictionary: {REQUIRED_CAMERA_PARAMS_KEYS}')
```
Comment (jean-lucas) [inline nucleus/camera_params.py:80]: Added this check.

File: nucleus/metadata_manager.py [hunk 4]
```diff
@@ -31,11 +34,24 @@ def __init__(
...
-        return payload
+            payload = {"reference_id": ref_id, "metadata": meta}
+
+            if self.level.value == ExportMetadataType.DATASET_ITEMS.value:
+                camera_params = self._extract_camera_params(meta)
+                if camera_params:
+                    payload[CAMERA_PARAMS_KEY] = camera_params.to_payload()
+
+            payloads.append(payload)
+        return payloads
```
Comment (jean-lucas) [inline nucleus/metadata_manager.py:54]: this is the main change of this PR. We send `metadata` alongside `camera_params` in the payload.

**Review by drakejwong — APPROVED** [review]
lgtm ty!

**Pull request merged**